pkg/utils/gateway.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	minIPV4CIDRBlockSize = 28
+	maxIPV4CIDRBlockSize = 16
+)
+
+// ParseIPV4GatewayNetmaskError is used to indicate any error with parsing the
+// IPV4 address and the netmask of the ENI
+type ParseIPV4GatewayNetmaskError struct {
+	operation string
+	origin    string
+	message   string
+}
+
+func (err *ParseIPV4GatewayNetmaskError) Error() string {
+	return err.operation + " " + err.origin + ": " + err.message
+}
+
+func newParseIPV4GatewayNetmaskError(operation string, origin string, message string) error {
```

Comment:
I think this should be public so that other packages can reuse this error?

Reply:
This is not being used externally today. We can make it public if/when needed.